core/relayMsgs.go

```diff
@@ -57,6 +57,8 @@ func (r *RelayMsgs) Send(src, dst Chain) {
 
 	r.Succeeded = true
 
+	srcMsgIDs := []MsgID{}
```

Comment:
You should fix this as the following.
```
srcMsgIDs := make([]MsgID, len(r.Src))
```
If sending `r.Src[i]` fails, then `srcMsgIDs[i]` should remain `nil`.

Reply:
Fixed